src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseDataFileService.cs

```diff
@@ -403,10 +461,10 @@ private async Task<string> GetSubjectName(Guid releaseId, File file)
         }
 
         private async Task<Either<ActionResult, string>> ValidateSubjectName(Guid releaseId,
-            string subjectName,
+            string? subjectName,
             File? replacingFile)
         {
-            if (replacingFile == null)
+            if (replacingFile == null && subjectName != null)
```

Comment:
Just wondering if we should enforce here that the name is present. Seems like we allow subjectName to be null when it's not a replacement and that will throw an exception rather than a return a validation result. If that's dealt with as early as possible then I think we avoid the optional subjectName in fileUploadsValidatorService.ValidateSubjectName.